orchestrator/cli/migrate_workflows.py

```diff
@@ -0,0 +1,239 @@
+# So the relationship product_type <-> workflow has cardinality 1 <-> *.  You can have zero or more workflows for a product_type, but every workflow has 1 product_type.
+# This is not evident in the db schema and we need to manually maintain this relation via the code for it to be consistent.
+
+# The workflows, products, product_blocks, product_product_blocks and products_workflows tables are all managed by code and migration files
+# New workflows are added with migration files, so they can be removed with migration files as well
+# We can assume that production environments are always test envs in terms of migrations executed. It is therefore ok use workflow names in the migration files, since
+# the database should only contain workflow names added from previous migration files.
+
+# Workflows without a product associated (tasks) should not be removed
+
+logger = structlog.get_logger(__name__)
+
+T = TypeVar("T")
+
+
+def _noqa_print(s: str, **kwargs: Any) -> None:
+    print(s, **kwargs)  # noqa: T201
+
+
+def _enumerate_menu_keys(items: list) -> List[str]:
+    return [str(i + 1) for i in range(len(items))]
+
+
+def _prompt_user_menu(
+    options: Iterable[Tuple[str, T]], keys: Optional[List[str]] = None, repeat: bool = True
+) -> Optional[T]:
+    options_list = list(options)
+    keys = keys or _enumerate_menu_keys(options_list)
+    done = False
+    while not done:
+        for k, txt_v in zip(keys, options_list):
+            _noqa_print(f"{k}) {txt_v[0]}")
+        choice = get_user_input("? ")
+        if choice not in keys:
+            _noqa_print("Invalid choice")
+            done = not repeat
+        else:
+            return options_list[keys.index(choice)][1]
+    return None
+
+
+def _print_workflows_table(workflows: List[WorkflowTable]) -> None:
+    items = [(wf.name, wf.target, wf.description, wf.products[0].product_type) for wf in workflows if wf.products]
+    print_fmt(
+        tabulate(
+            items,
+            headers=["#", "name", "target", "description", "product_type"],
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+
+
+def _add_workflow(workflows: Dict[str, LazyWorkflowInstance], state: dict) -> dict:
+    print_fmt("\nAdd new workflow\n", flags=[COLOR.UNDERLINE])
+
+    if not workflows:
+        _noqa_print("No registered workflows found to add to the database")
+        return state
+
+    registered_product_types = [
+        row[0] for row in ProductTable.query.with_entities(ProductTable.product_type).distinct()
+    ]
+    if not registered_product_types:
+        _noqa_print("No registered product types found")
+        return state
+
+    _noqa_print("Which product type should the workflow be added to?")
+    product_type = _prompt_user_menu(
+        [*[(p, p) for p in registered_product_types], ("cancel", None)],
+        keys=[*_enumerate_menu_keys(registered_product_types), "q"],
```

Comment:
(personal preference) I find it easier to read when combining two lists, rather than unpacking one list into another 

```suggestion
        [(p, p) for p in registered_product_types] + [("cancel", None)],
        keys=_enumerate_menu_keys(registered_product_types) + ["q"],
```